definitions/features/foreman_proxy.rb

```diff
@@ -1,21 +1,58 @@
 class Features::ForemanProxy < ForemanMaintain::Feature
   metadata do
     label :foreman_proxy
+
+    confine do
+      file_exists?('/etc/dhcp/dhcpd.conf') &&
+        ForemanMaintain::Utils::SmartProxySSL.instance.ssl_certs_exists?
+    end
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :base_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
```

Comment:
I feel there are lots of DHCP related calls viz. `valid_dhcp_configs?`, `dhcp_req_pass?`, `syntax_error_exists`.
A `DHCP` utility class can be initialized here.